packages/kit/src/core/make_package/index.js

```diff
@@ -98,6 +97,63 @@ export async function make_package(config, cwd = process.cwd()) {
 	}
 }
 
+/**
+ * @param {string} filename
+ * @param {string} source
+ */
+async function transpileTs(filename, source) {
+	const ts = await tryLoadTs();
+	return ts.transpileModule(source, {
+		compilerOptions: loadTsconfig(filename, ts),
+		fileName: filename
+	}).outputText;
+}
+
+async function tryLoadTs() {
+	try {
+		return (await import('typescript')).default;
+	} catch (e) {
+		throw new Error(
+			'You need to install TypeScript if you want to transpile TypeScript files and/or generate type definitions'
+		);
+	}
+}
+
+/**
+ *
+ * @param {string} filename
+ * @param {import('typescript')} ts
+ */
+function loadTsconfig(filename, ts) {
+	let basePath = process.cwd();
+	const fileDirectory = path.dirname(filename);
+	let tsconfigFile = ts.findConfigFile(fileDirectory, ts.sys.fileExists);
```

Comment:
Should we also consider a possibility of a separate `tsconfig` in `src/lib` exclusive for `lib` stuff?

Reply:
This should work already, as the nearest tsconfig for each file is looked up (which could be optimized, but I don't think it's a bottleneck right now).